internal/sauceignore/matcher.go

```diff
@@ -0,0 +1,77 @@
+package sauceignore
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
+)
+
+const commentPrefix = "#"
+const sauceignore = ".sauceignore"
+
+// ReadIgnoreFile reads .sauceignore file and creates ignore patters if .sauceignore file is exists.
+func ReadIgnoreFile(path string) ([]Pattern, error) {
```

Comment:
Would be better to distinguish it a bit more:
```suggestion
func PatternsFromFile(path string) ([]Pattern, error) {
```

Also, wouldn't it be better to have a convenience function that can create a matcher from file?
Based on the usage that I see here, the caller has to always read the file here, deal with potential errors and then get a matcher.

All callers would have to go through the same process; lots of boilerplate code. Would be great if you could create a matcher directly from file 💪

Reply:
I created additional function `NewMatcherFromFile` that creates`Matcher` from file. I also left `NewMatcher`, because it's convenient for testing to create `Matcher` just by passing patterns instead of creating a file with patterns.